File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/data/daos/IWeaponDao.java
package com.dkellycollins.triggerfinger.data.daos;

import com.dkellycollins.triggerfinger.data.entity.IWeapon;

/**
 * Provides te ability to manage weapon data.
 */
public interface IWeaponDao extends IActivityDao {

    Iterable<IWeapon> retrieve();
    IWeapon retrieve(int id);

    int create(int collidableId, int timerId);

    void delete(int id);
}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/data/daos/IWeaponDao.kt
package com.dkellycollins.triggerfinger.data.daos

import com.dkellycollins.triggerfinger.data.entity.IWeapon

/**
 * Provides te ability to manage weapon data.
 */
interface IWeaponDao : IActivityDao {

    fun retrieve(): Iterable<IWeapon>
    fun retrieve(id: Int): IWeapon

    fun create(collidableId: Int, timerId: Int): Int

    fun delete(id: Int)
}
